src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -72,26 +102,48 @@ boot_comp_pgtbl_expand(size_t n_pte, pgtblcap_t pt, vaddr_t vaddr, struct cobj_h
 		if (tot % SERVICE_SIZE) n_pte++;
 	}
 
-	for (i = 0; i < n_pte; i++) {
-		if (!cos_pgtbl_intern_alloc(boot_info(), pt, vaddr, SERVICE_SIZE)) BUG();
-		/* Increment vaddr incase we loop again */
-		vaddr += SERVICE_SIZE;
+	for (i = 0; i < n_pte; i++, vaddr+=SERVICE_SIZE) {
+		if (!cos_pgtbl_intern_alloc(boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
 	}
 }
 
+#define RESMGR_UNTYPED_MEM_SZ (COS_MEM_KERN_PA_SZ / 2)
+#define BOOT_COMP_UNTYPED_SZ  (1<<24) /* 16MB */
+
 /* Initialize just the captblcap and pgtblcap, due to hack for upcall_fn addr */
 static void
 boot_compinfo_init(spdid_t spdid, captblcap_t *ct, pgtblcap_t *pt, u32_t vaddr)
 {
-	*ct = cos_captbl_alloc(boot_info());
+	struct cos_compinfo *compinfo  = boot_spd_compinfo_get(spdid);
+	struct cos_compinfo *boot_info = boot_spd_compinfo_get(0);
+
+	*ct = cos_captbl_alloc(boot_info);
 	assert(*ct);
-	*pt = cos_pgtbl_alloc(boot_info());
+	*pt = cos_pgtbl_alloc(boot_info);
 	assert(*pt);
 
-	new_comp_cap_info[spdid].defcompinfo = &new_defcompinfo[spdid];
-	new_comp_cap_info[spdid].compinfo    = cos_compinfo_get(&new_defcompinfo[spdid]);
-	cos_compinfo_init(cos_compinfo_get(new_comp_cap_info[spdid].defcompinfo),
-			*pt, *ct, 0, (vaddr_t)vaddr, BOOT_CAPTBL_FREE, boot_info());
+	cos_compinfo_init(compinfo, *pt, *ct, 0, (vaddr_t)vaddr, BOOT_CAPTBL_FREE, boot_info);
+	/*
+	 * if this is a resmgr, let it manage its share (ideally rest of system memory) of memory.
+	 * if there is no resmgr in the system, allow every component to manage its memory.
+	 */
+	if (resmgr_spdid) {
```

Comment:
I cannot think of any other alternatives for global variables. I can a static variable and write wrapper around resmgr_spdid. 

The conditional above doesn't do exactly what I want it to do. 
My conditional is saying, if there is a resource manager in the system, only then check about this spdid. Otherwise let each component manage its own resources.

Above conditional will fail that for every component that is not resource manager even in the presence of a resource manager in the system. 

I could simplify this way for clarity, I'll do that. 
```
if (spdid && spdid == resmgr_spdid) {
      /* this is a resource manager */
} else if (!resmgr_spdid) {
      /* there is no resource manger */
}
```